extensions/interactions/LogicProof/protractor.js

```diff
@@ -0,0 +1,54 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview End-to-end testing utilities for the LogicProof interaction.
+ */
+
+var objects = require('../../objects/protractor.js');
+
+var customizeInteraction = function(elem, assumption, target, defaultText) {
+  elem.element(by.id('logicQuestionAssumptions')).sendKeys(assumption);
+  elem.element(by.id('logicQuestionTarget')).sendKeys(target);
+  elem.element(by.id('logicQuestionProof')).sendKeys(defaultText);
+};
+
+var expectInteractionDetailsToMatch = function(elem) {
+  expect(
+    elem.element(by.tagName('oppia-interactive-logic-proof')).isPresent()
+  ).toBe(true);
+};
+
+var submitAnswer = function() {
+  // TODO: find a way to send keys to the codeRepl. Temporarily pass the
```

Comment:
codeRepl?

Reply:
oops.. I meant code-mirror element that is being used here. Will change that. I had tried [this](https://stackoverflow.com/questions/30552393/how-to-write-protractor-test-for-ui-codemirror-angularjs-directive) but this seems to work in the player but not in the preview tab (not sure why it is inconsistent).

Reply:
done